locations/container/src/main/java/org/apache/brooklyn/container/location/kubernetes/KubernetesLocation.java

```diff
@@ -186,13 +185,36 @@ public KubernetesMachineLocation obtain(Map<?, ?> flags) {
     @Override
     public void release(KubernetesMachineLocation machine) {
         Entity entity = validateCallerContext(machine);
+        reinitClientIfExpired(entity);
```

Comment:
I think this is the wrong place for this.  There are other places where we use the client e.g. obtain and undeploy.  We would need to call this function before each use of the client in all these places.  I wonder if it might be better to move this to the getClient function and replace all calls to the client field with calls to getClient?

Reply:
Yes, that would make sense, the reason I kept it here for now is because I need the entity to get the namespace.